### ISSUE TITLE
Disable 1 specific javascript only?

### ISSUE THREAD

**Issue by voordev** [puppeteer/puppeteer#2145]
Using latest version of puppeteer and chromium.

Would it be possible to disable only 1 certain javascript inside by rewriting:
```
request.url 
```

I tried doing it as follow but this results in my pages not loading correctly ( missing selectors )
```
 const page              = await browser.newPage();
 await page.setRequestInterception(true);

 page.on('request', request => {
        if (request.resourceType == 'script')
            if (request.url == 'https://domain/link/to/custom/javascript.js')
            request.url = '';
            console.log("Rewrote Javascript link");
            request.continue();
        if (request.resourceType == 'font')
            request.abort();
        else
            request.continue();
    });
```

My goal is to block this specific javascript inside the page.
I thought so by changing the specific request.url if url is javascript condition is met would do the trick.

```
{ AssertionError [ERR_ASSERTION]: No node found for selector: #mycustomdiv
```

**Comment by deepakshrma** [comment]
Seems like you have not read the document. It says request is read-only. You need to set interceptor for these kinds of modification.

https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#event-request

> The request object is read-only. In order to intercept and mutate requests, see 
> page.setRequestInterception.

**Comment by voordev** [comment]
@deepakshrma 
Yes I have done this forgot to add it in the code, updated.

When I start init my page.
```
const page              = await browser.newPage();
await page.setRequestInterception(true);
```

**Comment by deepakshrma** [comment]
Please see the document. It says once you enable the RequestInterception. You can abort. But can't modify the request. It is immutable data object.
**Refernce: https://github.com/GoogleChrome/puppeteer/blob/v0.12.0/docs/api.md#pagesetrequestinterceptionenabledvalue**
Try: 
**request.continue([overrides]): As you raised another issue :-D
https://github.com/GoogleChrome/puppeteer/blob/v0.12.0/docs/api.md#requestcontinueoverrides**

**Comment by voordev** [comment]
@deepakshrma 

Thank you for pointing out. 
If I use the page.setRequestInterceptionEnabled 
I get a is not a function exception. I found in documentation latest API it was renamed to 'await page.setRequestInterception(true);':
```
await page.setRequestInterceptionEnabled(true);
or
await page.setRequestInterceptionEnabled(true);
```
Thus fails, so the correct way should be like following ( correct?)
```
 await page.setRequestInterception(true);

    page.setUserAgent('Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_2) AppleWebKit/604.4.7 (KHTML, like Gecko) Version/11.0.2 Safari/604.4.7')
    page.on('request', request => {

        if (request.resourceType == 'script')
            if (request.url == 'www.domain.com/path/to/my/js.js')
                console.log("FOUND JS");
                request.continue(request.url = '');
        if (request.resourceType == 'font')
            request.abort();
        else
            request.continue();
    });

```

The problem is this making my page break and unable to load a certain selector.

I found the problem of puppeteer not working on selecting checkboxes in the 'real world example' is loading of excessive javascript. Disabling all javascript however is not an option. 

_When I save a local version of the 'real world' code I try to select multiple checkboxes it works correctly but not in the 'real world' version, I found out the jquery does not load correctly in the local version ( I did a browser save all, so I get the html + all sources in a subfolder_ However this not loading jquery is not a problem and seems to resolve my selection issues!
Thus I look for an easy way to disable jquery somehow.. the problem seems that with this code example it does not work correctly..

I think of a workaround, would it be possible to save online version of the 'real world' example to puppeteer sandbox ( html page ) and then using that to submit instead of the 'real world' version of the page? Since we have a session ( its behind a login ) it would work since we would be posting the local version of the page within max 30 seconds, thus holding its session variables inside the page as correct.

_Any suggestions with this problem would be highly appreciated, are there documented problems with local versions of html pages where puppeteer works and 'real world' examples where puppeteer fails? Since the code is correct we could state there is 'something' in the 'real world' page that blocks puppeteer from being able to do selections_
